cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt

```diff
@@ -90,6 +91,66 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
         }
     }
 
+    /**
+     * Translates a Python [`With`](https://docs.python.org/3/library/ast.html#ast.With) into a
+     * [Block].
+     *
+     * We return a Block to handle the with statement, following
+     * [PEP 343](https://peps.python.org/pep-0343/#specification-the-with-statement). The context
+     * managers `__enter__()` runs before the try block and `__exit__()` is always called later in
+     * the `finally` block to clean up, whether an exception occurs or not.
+     */
+    private fun handleWithStatement(node: Python.AST.With): Block {
+        val mainBlock = newBlock().implicit()
```

Comment:
`Block` is wrong here. Should be a try / catch, once the other PRs are merged.

Reply:
Which other PRs do you mean? We have to return multiple statements here and afaik this has to be wrapped in a single statement by using a `Block`. If we had a way to return multiple statements which are then all added to the surrounding block, it would be great but I don't see any PR aiming at this.